### DISCUSSION TITLE
How do I extract fields from a POST body using Hyper?

### DISCUSSION THREAD

**Discussion by powerseed** [hyperium/hyper#3580] [Q&A] [answered]
Could someone take a look at the following question please? Thanks!
https://stackoverflow.com/questions/78054802/how-do-i-extract-fields-from-a-post-body-using-hyper

**Comment by dswij** [comment]
As mentioned in the stackoverflow comment, you're asking for a deserialization process that's not handled by hyper. You'd want something like https://github.com/serde-rs/json.

Also, please ask in one place only. Having two place of discussions is redundant and confusing.

**Reply by powerseed** [reply]
But how do I convert the Request to a string? I need to do this first, then I can convert the string to a JSON.

**Reply by dswij** [reply, marked as answer]
https://stackoverflow.com/questions/43419974/how-do-i-read-the-entire-body-of-a-tokio-based-hyper-request/77989320#77989320

**Reply by dswij** [reply]
Also, quoting your statement in the SO post: 

> I don't know why in Hyper it is so painful.

I suspect that you're unaware that `hyper` **is** supposed to be low-level. If you want convenience, consider higher-level libraries like `Axum` or `reqwest`.

**Reply by powerseed** [reply]
I see. I didn't try that answer. 
Thanks~
